sample-apps/deepedit_heart/train_heart.py

```diff
@@ -0,0 +1,345 @@
+        AddRandomGuidanced(guidance='guidance', discrepancy='discrepancy', probability='probability', batched=True),
+        # apply gaussian to guidance and add them as new channel
+        AddGuidanceSignald(image='image', guidance='guidance', batched=True),
+        DiscardAddGuidanced(image='image', batched=True),
+        ToTensord(keys=('image', 'label'))
+    ])
+
+
+def get_post_transforms():
+    return Compose([
+        Activationsd(keys='pred', sigmoid=True),
+        AsDiscreted(keys='pred', threshold_values=True, logit_thresh=0.5)
+    ])
+
+
+def get_loader(args, pre_transforms):
+    # Reading images
+    data_dir = os.path.join(args.input)
+    with open(os.path.join(data_dir, 'dataset.json')) as file:
+        data = json.load(file)
+    datalist = [{"image": os.path.join(args.input, d['image']),
+                 "label": os.path.join(args.input, d['labels'][0])} for d in data['objects']]
+    total_d = len(datalist)
+    total_l = len(datalist)
+
+    # Creating dataloader
+    train_ds = PersistentDataset(datalist, pre_transforms, cache_dir=args.cache_dir)
+    train_loader = DataLoader(
+        train_ds,
+        batch_size=args.batch,
+        shuffle=True,
+        num_workers=0)
+    logging.info('Total Records used for Training is: {}/{}/{}'.format(len(train_ds), total_l, total_d))
+
+    return train_loader
+
+
+def create_trainer(args):
+    set_determinism(seed=args.seed)
+
+    device = torch.device("cuda" if args.use_gpu else "cpu")
+
+    pre_transforms = get_pre_transforms(args.roi_size, args.model_size)
+    click_transforms = get_click_transforms()
+    post_transform = get_post_transforms()
+
+    train_loader = get_loader(args, pre_transforms)
+
+    # define training components
+    network = get_network(args.network, args.channels, args.dimensions).to(device)
+
+    if args.resume:
+        logging.info('Loading Network...')
+        map_location = {"cuda:0": "cuda:0"}
+        network.load_state_dict(torch.load(args.model_filepath, map_location=map_location))
+
+    loss_function = DiceLoss(sigmoid=True, squared_pred=True)
+    optimizer = torch.optim.Adam(network.parameters(), args.learning_rate)
+    lr_scheduler = torch.optim.lr_scheduler.StepLR(optimizer, step_size=5000, gamma=0.1)
+
+    train_handlers = [
+        LrScheduleHandler(lr_scheduler=lr_scheduler, print_lr=True),
+        StatsHandler(tag_name="train_loss", output_transform=lambda x: x["loss"]),
+        TensorBoardStatsHandler(log_dir=args.output, tag_name="train_loss", output_transform=lambda x: x["loss"]),
+        CheckpointSaver(save_dir=args.output, save_dict={"net": network, "opt": optimizer, "lr": lr_scheduler},
+                        save_interval=args.save_interval * 2, save_final=True, final_filename='checkpoint.pt'),
+        # # To check inner generated images
+        # InnerIterSaver(),
+    ]
+
+    trainer = SupervisedTrainer(
```

Comment:
we can actually use base BaseSegmentation class (which can be renamed to BaseTrainer class) and still adopt deepgrow training over there.. lets dicuss